### ISSUE TITLE
health checks

### ISSUE THREAD

**Issue by yoshuawuyts** [buildkite/charts#7]
sometimes the docker socket can get unmounted for reasons; adding a health check can be beneficial:

``` yaml
          livenessProbe:
            initialDelaySeconds: 15
            timeoutSeconds: 1
            exec:
              command:
                - sudo
                - docker
                - ps
```
### See Also
- http://kubernetes.io/docs/user-guide/liveness/

**Comment by rimusz** [comment]
I'm going to move these charts to Helm v2 type, so then I will include health check for the docker socket. Thanks for the suggestion.

**Comment by yoshuawuyts** [comment]
Ah, just ran into another issue: something was going wrong with `apt-get install` and the thing ran into a corrupted state. It'd be ideal if we had manual handles to detect if a container is corrupted and then restart if it does. E.g.
- run a build
- if anything in the build goes wrong during setup, write something to a file
- externally we ping the file every n seconds to check if things are ok
- if n lines have been written to the file (or any other way of notifying it, w/e) we fail the health check and restart the container

This is a mechanism to cope with a situation where corrupted containers are failing, and we need to distinguish between builds failing (e.g. expected) and actual init failing (e.g. unexpected, possibly corrupted state). Happy to hear if there's a better way to do this; writing to a file was the simplest mechanism I could think of. Cheers!

**Comment by rimusz** [comment]
liveness probe got implemented in #25
